controllers/basket.controllers.js

```diff
@@ -0,0 +1,143 @@
+const { validationResult } = require('express-validator');
+const HttpError = require('../utils/http-error');
+
+const Basket = require('../models/basket.model');
+const Participants = require('../models/participant.model');
+
+/*
+function includes(baskets, basket) {
+    for(let i = 0; i < baskets.length; i++){
+        if(JSON.stringify(baskets[i]) === JSON.stringify(basket)) { return true; }
+    }
+
+    return false;
+}s
+*/
+
+async function getOwnerBaskets(req, res, next) {
+    if(!req.user){
+        return next(new HttpError('A user is not logged in.', 401));
+    }
+
+    const baskets = await Basket.find({ ownerId: req.user._id });
+    res.status(200).json(baskets);
+}
+
+async function getCoownerBaskets(req, res, next) {
+    if(!req.user){
+        return next(new HttpError('A user is not logged in.', 401));
+    }
+
+    const baskets = await (await Participants.find({ user: req.user._id }).populate('basket'))
+                                .map((doc) => {
+                                    console.log(new Date(Date.parse(doc.basket.expirationDate) - Date.now()).getDate())
+                                    doc.basket.isHot = (new Date(Date.parse(doc.basket.expirationDate) - Date.now()).getDate() < 7);
+                                    return doc.basket;        
+                                })
```

Comment:
do u need await (await ...)? You don't have any async operations in map, I think it can work with (await ...).map()

Reply:
Resolved in Olegs branch